### DISCUSSION TITLE
Option to copy a toolbar

### DISCUSSION THREAD

**Discussion by DNightDM** [chrisgurney/obsidian-note-toolbar#48] [Ideas]
Hi Chris

Let me start with saying how much I love the Note Toolbar. It is an absoluty amazing addition to my plugin collection.

I'm using the toolbar to create easy navigation, mainly on mobile, between the folders where I keep my player notes for the TTRPG games I am in. Each game has their own folder and a standard set of files that I maintain within. This means that many of my toolbars more or less contain the same "buttons" but the file paths within obviously changes.  

It could be really neat to be able to copy an already finished toolbar and then modify file paths/links, instead of having to build each toolbar from scratch.

**Comment by chrisgurney** [comment]
@DNightDM Thanks for the suggestion!

Can you tell me more about how the folders are different, with some examples?

And are you using Commands, File, or URI type items in these toolbars?

**Reply by DNightDM** [reply]
Hi Chris

As of right now, I am mostly using File type items, except for a single command. 

Here is what I currently have. The toolbar layout for Game A is my current default navigation. Game B and C have the same navigations but with some additions. Each toolbar are mapped to their own game folder. 

Being able to copy the toolbar for Game A and change/add/delete within it as needed could be amazing. 

![Toolbar A](https://github.com/chrisgurney/obsidian-note-toolbar/assets/163132388/953cf652-6a99-4ec2-8e71-10c0b38c8edd)

![Game A+B+C](https://github.com/chrisgurney/obsidian-note-toolbar/assets/163132388/7b42d71e-867b-4c97-b5f9-c30d9e0838eb)

Hope it makes sense

**Reply by chrisgurney** [reply]
Thank you for sharing!

Was going to suggest potentially using [variables](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Variables) (if any of this is stored in Properties), but this definitely looks more involved due to the additional items.

Copying toolbars definitely seems like a good request that solves your issue. I've added it to the roadmap.

(There's also a related enhancement request around [inheritance](https://github.com/chrisgurney/obsidian-note-toolbar/discussions/32) that may be interesting to you.)

**Reply by AniOhev** [reply]
+1

**Reply by DNightDM** [reply]
Thanks Chris.

If it wasn't for the additions and subtractions to the toolbars I would definitely have solved it with variables. And I still should those that overlap across the folders.

**Comment by FeralFlora** [comment]
I was thinking of something similar when setting up #79, namely that it would be nice to have the ability to reuse buttons / items from other toolbars.

**Comment by chrisgurney** [comment]
@DNightDM @AniOhev @FeralFlora @charle36 FYI the ability to copy toolbars and toolbar items is now available in the [1.11.0 release](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.11.0).
